pb_decode.h

```diff
@@ -43,6 +43,10 @@ struct pb_istream_s
 #ifndef PB_NO_ERRMSG
     const char *errmsg;
 #endif
+  #ifdef PB_ENABLE_MALLOC_CONTEXT
+  /* Note: we do not need a corresponding free() here, because we can (and do) call realloc with size==0 to free memory. */
```

Comment:
meta data might be keeped so it won't completely free ?

Reply:
All definitions of realloc I've seen fully free the object. See e.g. https://pubs.opengroup.org/onlinepubs/009695299/functions/realloc.html *"If size is 0 and ptr is not a null pointer, the object pointed to is freed."*

Reply:
Looks like C17/C23 deprecate the zero size.